contracts/exchange-wrappers/ZeroExV2MultiOrderExchangeWrapper.sol

```diff
@@ -0,0 +1,399 @@
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+pragma solidity 0.5.3;
+pragma experimental ABIEncoderV2;
+
+import { IExchange } from "../external/0x/v2/interfaces/IExchange.sol";
+import { LibFillResults } from "../external/0x/v2/libs/LibFillResults.sol";
+import { LibOrder } from "../external/0x/v2/libs/LibOrder.sol";
+import { SafeMath } from "../external/openzeppelin-solidity/contracts/math/SafeMath.sol";
+import { ExchangeWrapper } from "../interfaces/ExchangeWrapper.sol";
+import { MathHelpers } from "../lib/MathHelpers.sol";
+import { TokenInteract } from "../lib/TokenInteract.sol";
+
+
+/**
+ * @title ZeroExV2MultiOrderExchangeWrapper
+ * @author dYdX
+ *
+ * dYdX ExchangeWrapper to interface with 0x Version 2. Sends multiple orders at once. Assumes no
+ * fees.
+ */
+contract ZeroExV2MultiOrderExchangeWrapper is
+    LibFillResults,
+    LibOrder,
+    ExchangeWrapper
+{
+    using SafeMath for uint256;
+    using TokenInteract for address;
+
+    // ============ Constants ============
+
+    // number of bytes in the maxPrice data
+    uint256 constant PRICE_DATA_LENGTH = 64;
+
+    // number of bytes per (order+signature)
+    uint256 constant ORDER_DATA_LENGTH = 322;
+
+    // ============ Structs ============
+
+    struct TokenAmounts {
+        uint256 takerAmount;
+        uint256 makerAmount;
+    }
+
+    // ============ State Variables ============
+
+    // address of the ZeroEx V2 Exchange
+    address public ZERO_EX_EXCHANGE;
+
+    // address of the ZeroEx V2 ERC20Proxy
+    address public ZERO_EX_TOKEN_PROXY;
+
+    // ============ Constructor ============
+
+    constructor(
+        address zeroExExchange,
+        address zeroExProxy
+    )
+        public
+    {
+        ZERO_EX_EXCHANGE = zeroExExchange;
+        ZERO_EX_TOKEN_PROXY = zeroExProxy;
+    }
+
+    // ============ Public Functions ============
+
+    function exchange(
+        address /* tradeOriginator */,
+        address receiver,
+        address makerToken,
+        address takerToken,
+        uint256 requestedFillAmount,
+        bytes calldata orderData
+    )
+        external
+        returns (uint256)
+    {
+        // parse all order data
+        validateOrderData(orderData);
+        TokenAmounts memory priceRatio = parseMaxPriceRatio(orderData);
+        Order[] memory orders = parseOrders(orderData, makerToken, takerToken);
+        bytes[] memory signatures = parseSignatures(orderData);
+
+        // ensure that the exchange can take the tokens from this contract
+        ensureAllowance(
+            takerToken,
+            ZERO_EX_TOKEN_PROXY,
+            requestedFillAmount
+        );
+
+        // do the exchange
+        FillResults memory totalFillResults = IExchange(ZERO_EX_EXCHANGE).marketSellOrdersNoThrow(
```

Comment:
Can you explain what “no throw” means on the 0x contract? What does it not throw on? Will it not throw even if the sell fails?

Reply:
It won't revert if it fails to fill one of the orders (like it keeps continuing no matter what).

So yes it won't throw even if all sells fail